src/main/java/com/google/sps/servlets/MatchServlet.java

```diff
@@ -25,40 +29,58 @@
   public void init(ServletConfig config) throws ServletException {
     super.init(config);
 
-    NonPersistentMatchRepository repository = new NonPersistentMatchRepository();
-    User testUser = repository.addTestData();
+    //won't need ServletContext once we switch to PersistentMatchRepository
+    MatchRepository matchRepository = new NonPersistentMatchRepository();
+    getServletContext().setAttribute("matchRepository", matchRepository);
```

Comment:
I noticed that we may have some shared logic between the MatchServlet and the NewMatchQueryServlet both with managing the repositories and with getting the logged in user. Maybe it would make sense to create a new class named MatchContext which manages the repositories, can get the logged in User, and can be shared between the two servlets?

Let me know what you think and feel free to defer this to a future pull request.

Reply:
Just submitted this in a separate PR!